Public/Src/Engine/Processes/FileAccessPolicy.cs

```diff
@@ -81,6 +81,19 @@ public enum FileAccessPolicy : ushort
         /// </remarks>
         AllowRealInputTimestamps = 0x200,
 
+        /// <summary>
+        /// Override writes allowed by policy based on file existence checks. 
+        /// </summary>
+        /// <remarks>
+        /// Override writes allowed by policy based on file existence checks. 
+        /// Used mainly in the context of shared opaques, where the whole cone under the opaque root is write-allowed by policy (except known inputs).
+        /// This policy makes sure that writes on undeclared inputs that fall under the write-allowed cone are flagged as DFAs.
+        /// The way to dermine undeclared inputs is based on file existence: if a pip tries to write into a file - allowed by policy - but
+        /// that was not created by the pip (i.e. the file was there before the first write), then it is a write on an undeclared input

```

Comment:
>the first write [](start = 76, length = 15)

Is this the first write by the same pip? Or could it be the first write by another pip?

Reply:
Before the first write by this pip. I.e. the pip didn't create the file.

---
In reply to: [299179727](https://github.com/microsoft/BuildXL/pull/536#discussion_r299179727) [](ancestors = 299179727)

Reply:
Can you complete the sentence in the comment "the file was there before the first write by this pip" ?

---
In reply to: [299186944](https://github.com/microsoft/BuildXL/pull/536#discussion_r299186944) [](ancestors = 299186944,299179727)

Reply:
Sure

---
In reply to: [299195913](https://github.com/microsoft/BuildXL/pull/536#discussion_r299195913) [](ancestors = 299195913,299186944,299179727)